src/page/blog/ProgrammingLanguageDefense.js

```diff
@@ -0,0 +1,285 @@
+                                </li>
+                                <li>Improved ergonomics: Better IDEs, better APIs, etc.</li>
+                            </ul>
+
+                            I don't think we are anywhere near the point where programming languages are as good as they
+                            are every going to get. I think we are still in the infancy of the arc of programming
+                            languages.
+                        </p>
+
+                        <h5>All programming language are the same</h5>
+
+                        <p>
+                            It strongly disagree. I think we are experiencing a period of programming language
+                            fragmentation after a long period of consolidation. For the last 15-years or so, the
+                            industry was dominated by C, C++, C# and Java. The market share of these programming
+                            languages were always increasing. I view this as a consolidation phase where there was a
+                            sense that object-oriented programming languages such as C# and Java were the way to go.
+                        </p>
+
+                        <p>
+                            Today that is no longer the case. Instead, the ecosystem is much more diverse and stronger
+                            for it. We have Rust. We have Scala. We have Go, Python, and JavaScript. We have Crystal and
+                            Nim. We have TypeScript. We are in a period of fragmentation. After a decade (or more) of
+                            OO-ossification we are finally in a new and exciting period!
+                        </p>
+
+                        <p>
+                            If history repeats itself then at some point we will enter a new period of consolidation. I
+                            am not sure who will "win", but whoever the winners are, I am sure that they will be
+                            "better" languages than those of the last cycle.
+                        </p>
+
+                        <p>
+                            (Addendum: Of course it is true that when it comes to hobby programming languages many are
+                            either purely functional or purely object-oriented programming languages. But that makes
+                            sense, of course hobbyist compiler hackers are interested in tinkering with the tools they
+                            use at their day job. This does not mean all programming languages are the same. Rather it
+                            is a question of better communication, which I will get to.)
+                        </p>
+
+                        <h5>Go / Python / Haskell is better</h5>
+
+                        <p>
+                            Yes, hopefully there is some progress. In the sense that the programming languages of
```

Comment:
See below about my interpretation of this section as a whole.
I'm confused about how to follow this rebuttal.

1. I introduce MattLang
2. Person says Haskell is better (meaning better than MattLang I assume?)
3. You say "Yes [meaning 'yes Haskell is better'?], hopefully there is some progress (meaning MattLang innovates somehow?)" Apparent contradiction there. 

> PLs of today better than PLs of yesterday, but they may not remain better

Is this saying that the PLs of yesterday will catch back up to those of today?